### ISSUE TITLE
Handling Trailing Space with a useCombobox Autocomplete

### ISSUE THREAD

**Issue by neilchaudhuri** [downshift-js/downshift#1247]
- `downshift` version: 6.1.0
- `node` version: 12.18.3
- `npm` (or `yarn`) version: 6.14.6

**Relevant code or config**

```
setFilteredOptions(
					(filter && filter(inputValue)) ||
						options.filter(o => o.label.toLowerCase().indexOf(inputValue.toLowerCase()) >= 0)
				)
```

**What you did**:

Type in "united "

Note the space on the end.

**What happened**:

![image](https://user-images.githubusercontent.com/1239158/110522808-52ca8b00-80df-11eb-90b5-3e1fe36715e0.png)

![image](https://user-images.githubusercontent.com/1239158/110522830-578f3f00-80df-11eb-9a7f-50a345a34c49.png)

**Reproduction repository**: N/A

**Problem description**:

Not a bug. Just a question. I have an autocomplete with `useCombobox` item that has spaces--in this case "United States."

When I type "united" things work as expected as the option is filtered and the dropdown expands

When I type "united " (with a space on the end) the filtered option is still correct but the dropdown collapses.

When I type "united s" things work as expected as the option is filtered and the dropdown expands

I have tried a lot of different ways to deal with keeping the dropdown open with that trailing space but to no avail. Any tips?

**Suggested solution**: N/A

**Comment by silviuaavram** [comment]
inputValue should also contain the space, can you check this please? not sure if it's a problem with downshift here, but if you dig into it and we actually have a bug, please reopen this ticket. Thanks very much!

From the docsite:

![Screenshot 2021-03-11 at 09 24 12](https://user-images.githubusercontent.com/11275392/110750784-ad80f580-824b-11eb-8868-4c3f65f479df.png)

![Screenshot 2021-03-11 at 09 25 52](https://user-images.githubusercontent.com/11275392/110750855-cf7a7800-824b-11eb-9b16-d218ad10ca9f.png)

**Comment by neilchaudhuri** [comment]
Thanks for looking into this. I should have mentioned that I have already checked `onInputValueChange` and `stateReducer`. The latter is "one character behind" as the state changes, but both do capture the space. And as I did mention before, the filter `onInputValueChange` works properly in that the choices are reduced, in my example, to "United States." 

The only problem is that when the last character typed is a space as in "united " the once opened list closes and only reopens  when I append the "s." In other words, _the behavior is only an issue with a trailing space_. The intervening space is just fine when it happens eventually.

The only difference I can tell in your code is your filter function. Mine is this:

`options.filter(o => o.label.toLowerCase().indexOf(inputValue.toLowerCase()) >= 0)`

**Comment by silviuaavram** [comment]
<img width="707" alt="Screenshot 2021-03-19 at 15 13 17" src="https://user-images.githubusercontent.com/11275392/111904609-729f6e80-8a50-11eb-8f47-e35e07a7543b.png">

The docs are public at https://www.downshift-js.com/use-combobox so check them out :D

**Comment by OctaveLaventure** [comment]
In case it helps someone:
 I faced the same issue today. I analyzed the state updates with
```
 useCombobox({
    [...]
    onStateChange: console.debug,
  })
```
and noticed that a space keydown event was triggering a state update with the type `__togglebutton_click__` right after the normal `__input_change__` one. I then found that my input was wrapped in another clickable element, on which a click event could be triggered by the space key. It triggered the toggle-on-click of the autocomplete (which is a normal behavior) and thus closed the dropdown list each time I typed a space.
cc @neilchaudhuri
